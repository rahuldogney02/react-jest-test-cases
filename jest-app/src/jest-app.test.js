import {fireEvent, render , screen} from '@testing-library/react';
import MyJestApp from './jest-app';

const titleValue= 'freepik images';

test('test MyJestApp', ()=>{
    render(<MyJestApp />);
    const  text = screen.getByText('Hello Jest test case');
    const  caseInsentive = screen.getByText(/hello jest test case/i);
    const title = screen.getByTitle(titleValue);
    expect(caseInsentive).toBeInTheDocument();
    expect(text).toBeInTheDocument();
    expect(title).toBeInTheDocument();

})

test('Testing of Input box',()=>{
    render(<MyJestApp/>);
    let checkInput = screen.getByRole('textbox');
    let checkPlaceHolder = screen.getByPlaceholderText('Enter first name')
    expect(checkInput).toBeInTheDocument();
    expect(checkPlaceHolder).toBeInTheDocument();
    expect(checkInput).toHaveAttribute("name","firstname")
    expect(checkInput).toHaveAttribute("type","text")

})

describe("UI test cases in Group",()=>{
    test('test one MyJestApp', ()=>{
        render(<MyJestApp />);
        const  text = screen.getByText('Hello Jest test case');
        const  caseInsentive = screen.getByText(/hello jest test case/i);
        const title = screen.getByTitle(titleValue);
        expect(caseInsentive).toBeInTheDocument();
        expect(text).toBeInTheDocument();
        expect(title).toBeInTheDocument();
    
    })
    // test.only('Testing of two Input box',()=>{
    //     render(<MyJestApp/>);
    //     let checkInput = screen.getByRole('textbox');
    //     let checkPlaceHolder = screen.getByPlaceholderText('Enter first name')
    //     expect(checkInput).toBeInTheDocument();
    //     expect(checkPlaceHolder).toBeInTheDocument();
    //     expect(checkInput).toHaveAttribute("name","firstname")
    //     expect(checkInput).toHaveAttribute("type","text")
    
    // })
    test.skip('Skip Input box',()=>{
        render(<MyJestApp/>);
        let checkInput = screen.getByRole('textbox');
        let checkPlaceHolder = screen.getByPlaceholderText('Enter first name')
        expect(checkInput).toBeInTheDocument();
        expect(checkPlaceHolder).toBeInTheDocument();
        expect(checkInput).toHaveAttribute("name","firstname")
        expect(checkInput).toHaveAttribute("type","text")
    
    })
})

test('input field fireEvent',()=>{
    render(<MyJestApp/>);

    let inputName = screen.getByRole('textbox');
    fireEvent.change(inputName,{target:{value:'113'}});
    expect(inputName.value).toBe("113Test")
})