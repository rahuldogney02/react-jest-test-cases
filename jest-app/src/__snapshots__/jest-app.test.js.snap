// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="border: 2px dashed black; padding: 10px; background: black; color: red; display: flex;"
      >
        <h1>
          Hello Jest test case
        </h1>
        <p>
          DeprecationWarning: The \`punycode\` module is deprecated. Please use a userland alternative instead.
        </p>
        <img
          height="100px"
          src="https://img.freepik.com/free-psd/black-ink-splatter-abstract-art-design-graphic_84443-35548.jpg?ga=GA1.1.1057615231.1738340637&semt=ais_hybrid"
          title="freepik images"
          width="100px"
        />
      </div>
      <div
        style="border: 2px dashed black; padding: 40px; background: gray; color: blue; display: flex;"
      >
        <div>
          <label>
            First Name
          </label>
          <input
            name="firstname"
            placeholder="Enter first name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div>
        <button>
          Update Data
        </button>
        <h1 />
      </div>
      <h2>
        Snapshot testing
      </h2>
    </div>
  </body>,
  "container": <div>
    <div
      style="border: 2px dashed black; padding: 10px; background: black; color: red; display: flex;"
    >
      <h1>
        Hello Jest test case
      </h1>
      <p>
        DeprecationWarning: The \`punycode\` module is deprecated. Please use a userland alternative instead.
      </p>
      <img
        height="100px"
        src="https://img.freepik.com/free-psd/black-ink-splatter-abstract-art-design-graphic_84443-35548.jpg?ga=GA1.1.1057615231.1738340637&semt=ais_hybrid"
        title="freepik images"
        width="100px"
      />
    </div>
    <div
      style="border: 2px dashed black; padding: 40px; background: gray; color: blue; display: flex;"
    >
      <div>
        <label>
          First Name
        </label>
        <input
          name="firstname"
          placeholder="Enter first name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div>
      <button>
        Update Data
      </button>
      <h1 />
    </div>
    <h2>
      Snapshot testing
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
